code/_helpers/visual_filters.dm

```diff
@@ -26,8 +29,9 @@
 	UPDATE_OO_IF_PRESENT
 
 /atom/movable/proc/get_filter(filter_name)
-	if(filter_data && filter_data[filter_name])
-		return filters[filter_data.Find(filter_name)]
+	var/filter_index = filter_data?.Find(filter_name)
+	if(filter_index > 0 && filter_index <= length(filters))
```

Comment:
I was getting a bunch of runtimes with filters that I'm not sure how to debug or explain. This check at least stopped it runtiming in the middle of unrelated code.